packages/base/src/components/Link/Link.base.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import { ILinkProps, ILinkSlots } from "./Link.types";
+import { useLink } from "./useLink";
+
+export const LinkBase: React.FunctionComponent<ILinkProps> = (
+  props: ILinkProps
+) => {
+  const { children, href, slots } = props;
+    const tag = _deriveRootType(href);
```

Comment:
```suggestion
    const tag = href ? 'a' : 'button';
```
function name is longer than the function return :)

If this root type approach is used everywhere, and the idea is consistency and forward thinking, it's fine, but for simplicity, keeping these together makes reading much easier.

Reply:
I agree with you, I'll change this.